### DISCUSSION TITLE
在insert directives的時候就先打開directive.tcl導致後面的port沒法insert directive

### DISCUSSION THREAD

**Discussion by holyuming** [bol-edu/HLS-SOC-Discussions#72] [Q&A] [answered]
![1](https://github.com/bol-edu/HLS-SOC-Discussions/assets/85299700/4d3c9f07-aee6-4344-8ac1-5fcca8aacad3)

可以看到右邊有port的資訊可以insert directive
但是在我對fir_n11_maxi做insert directive之後，我去點開左邊panel的solution1/constraints/directives.tcl，如下:
![image](https://github.com/bol-edu/HLS-SOC-Discussions/assets/85299700/c156f970-402a-4367-b8ef-9da6378323ae)
非常合理，因為我只有對fir_n11_maxi加入directive，可是就在這時，我想回去對原本在右邊剩下的input & output port加入directives，
![image](https://github.com/bol-edu/HLS-SOC-Discussions/assets/85299700/a42b96cd-b6f6-48e6-84fa-3d864945e16d)
他們卻不見了。

我原本想到的方法是直接去directives.tcl
加入剩下的directive
可是
![image](https://github.com/bol-edu/HLS-SOC-Discussions/assets/85299700/5270d62a-edda-4a85-ad5e-f54ad8b2894c)
這個檔案卻寫說
Please do not edit it
請問有人有解嗎?
感謝

**Comment by b3nsonchang** [comment, marked as answer]
我也有遇到這個問題，只好把檔案重設定一遍，中間不要去開directives.tcl

**Comment by ghost** [comment]
做insert directive中間應該還是可以打開directives.tcl的，我在完成fir_n11_ma的設定後，有先打開directives.tcl查看內容，但我有將視窗中directives.tcl檔案關掉再繼續做其他port的insert directive，有成功完成模擬，我猜想應該是將檔案開啟，EDA tool 無法對directives.tcl進行編輯。
![1](https://github.com/bol-edu/HLS-SOC-Discussions/assets/119398292/cd2210be-d822-49f4-89c8-bda48b2812b9)
